ui/src/message_popup/renderers/audio_response_summary.jsx

```diff
@@ -13,22 +13,24 @@ export default React.createClass({
   displayName: 'AudioResponseSummary',
 
   propTypes: {
-    gameSession: React.PropTypes.object.isRequired,
+    audioResponses: React.PropTypes.array.isRequired,
   },
 
   render() {
+
     const audioResponses = [];
-    const reviewedQuestions = {};  // Keeps track of reviewedQuestions to ensure we only show the last response if the user does multiple retries.
-    this.props.gameSession.audioResponses.map((arObj, i) => {
-      var obj = {};
-      obj.audioUrl = arObj.blobUrl;
-      obj.questionText = arObj.question.text;
-      var questionId = arObj.question.id;
-      if(questionId in reviewedQuestions) {
-        audioResponses[reviewedQuestions[questionId]] = obj;
+    var lastQuestionId = null;  // For detecting questions with more than one response
```

Comment:
Thanks for explaining!  For keeping the last one I think that'd be the same as adding in some `reverse`s with `_.uniq`.

For the change here, you could considering using other higher-level functions to express the intent more directly:
```
responsesByQuestion = _.groupBy(audioResponses, ...);
questionsWithResponses = _.map(responsesByQuestion, ...);
```

Or you could split this code out into a method and name it something that indicates what it will return.  This would more directly express that the shape of the data is changing - here this chunk of code yields the same variable name `audioResponses`, even though that value has a different shape of data than the original `this.props.audioResponses`.